lib/ansible/modules/cloud/cloudstack/cs_firewall.py

```diff
@@ -17,123 +17,138 @@
 description:
     - Creates and removes firewall rules.
 version_added: '2.0'
-author: "René Moser (@resmo)"
+author: René Moser (@resmo)
 options:
   ip_address:
     description:
       - Public IP address the ingress rule is assigned to.
-      - Required if C(type=ingress).
+      - Required if I(type=ingress).
+    type: str
   network:
     description:
       - Network the egress rule is related to.
-      - Required if C(type=egress).
+      - Required if I(type=egress).
+    type: str
   state:
     description:
       - State of the firewall rule.
+    type: str
     default: present
     choices: [ present, absent ]
   type:
     description:
       - Type of the firewall rule.
+    type: str
     default: ingress
     choices: [ ingress, egress ]
   protocol:
     description:
       - Protocol of the firewall rule.
-      - C(all) is only available if C(type=egress).
+      - C(all) is only available if I(type=egress).
+    type: str
     default: tcp
     choices: [ tcp, udp, icmp, all ]
   cidrs:
     description:
       - List of CIDRs (full notation) to be used for firewall rule.
       - Since version 2.5, it is a list of CIDR.
+    type: list
     default: 0.0.0.0/0
     aliases: [ cidr ]
   start_port:
     description:
       - Start port for this rule.
-      - Considered if C(protocol=tcp) or C(protocol=udp).
+      - Considered if I(protocol=tcp) or I(protocol=udp).
+    type: int
     aliases: [ port ]
   end_port:
     description:
-      - End port for this rule. Considered if C(protocol=tcp) or C(protocol=udp).
-      - If not specified, equal C(start_port).
+      - End port for this rule. Considered if I(protocol=tcp) or I(protocol=udp).
+      - If not specified, equal I(start_port).
+    type: int
   icmp_type:
     description:
       - Type of the icmp message being sent.
-      - Considered if C(protocol=icmp).
+      - Considered if I(protocol=icmp).
+    type: int
   icmp_code:
     description:
       - Error code for this icmp message.
-      - Considered if C(protocol=icmp).
+      - Considered if I(protocol=icmp).
+    type: int
   domain:
     description:
       - Domain the firewall rule is related to.
+    type: str
   account:
     description:
       - Account the firewall rule is related to.
+    type: str
   project:
     description:
       - Name of the project the firewall rule is related to.
+    type: str
   zone:
     description:
       - Name of the zone in which the virtual machine is in.
       - If not set, default zone is used.
+    type: str
   poll_async:
     description:
       - Poll async jobs until job has finished.
-    default: true
     type: bool
+    default: yes
```

Comment:
The common Ansible standard seems to be "yes" and "no" for booleans, but it looks like the recommendation for YAML 1.2 will be a limitation to "true" and "false" (presumably to be closer to JSON): https://yaml.org/spec/1.2/spec.html#id2804923 and https://yaml.org/spec/1.2/spec.html#id2803629

In light of that, does it make sense to insist on "yes" and "no"?
I'm not saying this must be changed, as I'm sure that Ansible will keep supporting those values even if moving to YAML 1.2 at some point. Just a general point.

Reply:
I was always surprised to see this `yes/true` flag value that could cause some sort of confusion in how to set boolean flag. Happy to see that it's moving to a clearer definition.

Reply:
I made these changes for consistency with the Ansible standard and doc_fragments. It will also be less confusing if it's `yes/no` for all modules.
But you're right, it perhaps doesn't make sense.

Reply:
values in choices and default for bool should be `yes/no` according doc of ansible. it is a convention by Ansible. `required` should be true/false as it is a flag to show it is required.

Reply:
Thanks for the clarification, @resmo . I suspected there was a guideline for that.